File: src/sdk-reference/kuzzle/query/snippets/query.cpp
try {
  kuzzle_request request = {0};
  request.controller = "document";
  request.action = "create";
  request.index = "nyc-open-data";
  request.collection = "yellow-taxi";
  request.id = "my-custom-document-id";
  request.body = "{\"trip_distance\": 4.23, \"passenger_count\": 2}";

  query_options options = {0};
  options.refresh = "wait_for";

  kuzzle_response* response = kuzzle->query(&request, &options);

  if (response->status == 200) {
    std::cout << "Document created" << std::endl;
  }
}
catch (kuzzleio::UnauthorizedException& e) {
  std::cerr << "You are not connected" << std::endl;
}
catch (kuzzleio::KuzzleException& e) {
  std::cerr << e.getMessage() << std::endl;
}

File: src/sdk-reference/kuzzle/query/snippets/query.java
try {
  KuzzleRequest request = new KuzzleRequest();
  request.setController("document");
  request.setAction("create");
  request.setIndex("nyc-open-data");
  request.setCollection("yellow-taxi");
  request.setId("my-custom-document-id");
  request.setBody("{\"trip_distance\": 4.23, \"passenger_count\": 2}");

  QueryOptions options = new QueryOptions();
  options.setRefresh("wait_for");

  KuzzleResponse response = kuzzle.query(request, options);

  if (response != null) {
    System.out.println("Document created");
  }
}
catch (UnauthorizedException e) {
  System.err.println("You are not connected");
}
catch (KuzzleException e) {
  System.err.println(e.getMessage());
}
